bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java

```diff
@@ -71,11 +80,15 @@
     final long ledgerId;
     long lastAddPushed;
     volatile long lastAddConfirmed;
+    volatile long explicitLastAddConfirmed;
+    volatile long piggyBackedLastAddConfirmed;
+
     long length;
     final DigestManager macManager;
     final DistributionSchedule distributionSchedule;
 
     final RateLimiter throttler;
+    private ScheduledExecutorService lastAddConfirmedUpdateScheduler = null;
```

Comment:
ExplicitLAC is bese-effort from update/publish. It should not break any functionality if it gets delayed. So if lac reaches bookie outof order, it is ok.